paddle/fluid/memory/allocation/retry_allocator_test.cc

```diff
@@ -93,6 +97,50 @@ TEST(RetryAllocator, RetryAllocator) {
   }
 }
 
+class DummyAllocator : public Allocator {
+ public:
+  bool IsAllocThreadSafe() const override { return true; }
+
+ protected:
+  Allocation *AllocateImpl(size_t size) override {
```

Comment:
Great dependency injection for test! Nothing to change. Just applause.

Reply:
Thanks!